examples/data_preprocessing/[WIP]_chexpert_dataset/prepare_CheXpert_data.py

```diff
@@ -0,0 +1,430 @@
+    uncertain (-1.0)
+    not-mentioned (blank), which is read in as NaN
+"""
+
+training_set.head(5)
+
+print("Number of observations in training set:", training_set.shape[0])
+
+### Get validation data
+
+"""
+first 5 entries of validation set
+note that this dataset has 2 possible labels:
+    positive (1.0)
+    negative (0.0)
+"""
+
+validation_set = pd.read_csv("valid.csv")
+
+validation_set.head(5)
+
+print("Number of observations in validation set:", validation_set.shape[0])
+
+### Collect statistics
+
+"""
+number of non-NaN labels in the training set
+"""
+
+training_labels = training_set.iloc[:, -13:-1]
+labels_per_row = training_labels.count(axis=1) # number of non-NaN labels per row in the training set
+
+vals = pd.DataFrame(labels_per_row.value_counts())
+
+# make a table
+val_list = [(i, vals[0][i]) for i in vals.index]
+    
+print(tabulate(val_list, headers=["Number of non-NaN labels", "Number of datapoints"]))
+
+"""
+label distribution in the training set
+"""
+
+val_list = []
+for cond in training_labels.columns:
+    vals = np.array(pd.Categorical(training_labels[cond], categories=[-1.0, 0.0, 1.0]).value_counts().sort_index(ascending=True))
+    val_list.append([cond, vals[0], vals[1], vals[2]])
+
+print("Label distribution in the training set:", "\n")
+print(tabulate(val_list, headers=["Pathology", "-1.0", "0.0", "1.0"]))
+
+"""
+label distribution in the validation set
+"""
+
+validation_labels = validation_set.iloc[:, -13:-1]
+
+val_list = []
+for cond in validation_labels.columns:
+    vals = np.array(pd.Categorical(validation_labels[cond], categories=[0.0, 1.0]).value_counts().sort_index(ascending=True))
+    val_list.append([cond, vals[0], vals[1]])
+        
+print("Label distribution in the validation set:", "\n")
+print(tabulate(val_list, headers=["Pathology", "0.0", "1.0"]))
+
+## Image preprocessing
+
+# paths to training images
+image_paths_train = [os.path.join(path[: path.find("CheXpert-v1.0-small")], "CheXpert-v1.0-small", p) for p in training_set["Path"]]
+
+# paths to validation images
+image_paths_valid = [os.path.join(path[: path.find("CheXpert-v1.0-small")], "CheXpert-v1.0-small", p) for p in validation_set["Path"]]
+
+# sample image from training set
+sample_image = Image.open(image_paths_train[0]).convert('RGB')
+plt.imshow(sample_image)
+plt.show()
+print("Dimensions of image:", sample_image.size)
+
+# define list of transformations that should be applied to the images
+transform_list = transforms.Compose([
+    transforms.Resize((224, 224)),
+    transforms.ToTensor(),
+    transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225]) # normalization from ImageNet
+    ])
+
+# class used for data loading and image preprocessing
+# inspirations for class and transformations above:
+     # https://github.com/gaetandi/cheXpert/blob/master/cheXpert_final.ipynb
+     # https://github.com/Stomper10/CheXpert/blob/master/CheXpert_DenseNet121_FL.ipynb
+     
+class CheXpertDatasetProcessor():

```

Comment:
I think it makes a lot of sense to write a Processor class, as you need a lot of information (labels) for each image. 
In general I think this is totally fine, but the functional style is more used in Big Data (cause easier to parallelize etc.).
Like:
```
1) df = transform_labels(df, **kwargs)
2) transform_images (path, df, **kwargs):
          for img in os.list(path):   # parallelizable
                  transform_img(path, img, df, **kwargs)
```